### PR TITLE
MRG, BUG: Fix error message listing of types

### PR DESCRIPTION
**Pull request by larsoner**
The type listing was wrong in #9447, would have made @sappelhoff's life easier in https://github.com/mne-tools/mne-bids/pull/807

### CODE REVIEW

**Comment by larsoner** [comment]
Azure failure should be fixed by #9476

**Review by hoechenberger — APPROVED** [review]
Small nitpick you may opt to ignore, otherwise LGTM

**Review by sappelhoff — APPROVED** [review]
This is great and would have simplified my search by a lot! Thanks.

File: mne/channels/montage.py [hunk 1]
```diff
@@ -894,14 +894,21 @@ def _backcompat_value(pos, ref_pos):
...
+            channel_type(info, non_picks[ii]) for ii in extra)))
         names = [non_names[ii] for ii in extra]
         warn(f'Not setting position{_pl(extra)} of {len(extra)} {types} '
              f'channel{_pl(extra)} found in montage:\n{names}\n'
-             'Consider setting the channel types to be of EEG/sEEG/ECoG/DBS '
-             'using inst.set_channel_types before calling inst.set_montage.')
+             'Consider setting the channel types to be of '
+             f'{_docdict["montage_types"]} '
+             'using inst.set_channel_types before calling inst.set_montage, '
+             'or omit these channels when creating your montage.')
```
[the remark this answers is not in the document]
Comment (larsoner) [reply mne/channels/montage.py:911]: I'd rather leave the `f` off when not needed, unlike other inconsistencies (that I also dislike) this one actually adds information, namely which lines have interpreted code and which don't. You can also figure that out from actually reading the line, but the preceding f not being on a line immediately tells you that you don't need to bother thinking about it

**Pull request merged**